Adding how to install service brokers

bergerhoffer opened the pull request:
@openshift/team-documentation For peer review please.

Preview: 

* http://file.rdu.redhat.com/~ahoffer/2019/OSDOCS-176-install-sb/applications/service_brokers/installing-service-catalog.html
* http://file.rdu.redhat.com/~ahoffer/2019/OSDOCS-176-install-sb/applications/service_brokers/installing-template-service-broker.html
* http://file.rdu.redhat.com/~ahoffer/2019/OSDOCS-176-install-sb/applications/service_brokers/installing-ansible-service-broker.html

johnkim76 commented:
@bergerhoffer, Please remove the the `Create a cluster role binding` step for TSB, since I've verified that is not required.

Also, the `click show community operators` seemed to have changed.  That option does not seem to exist, but when the user selects the operator to install, they are greeted with a popup warning, which they'll need to click `continue` in order to proceed with installing the operator

bergerhoffer commented:
Thanks @johnkim76! I've updated the docs if you can take another look. I made the following changes:

1. Removed the operator group step (TSB/ASB).
2. Removed the cluster role binding step (TSB).
3. Updated starting ASB/TSB procedure to use the UI.
4. Updated the community operator step (ASB/TSB).
5. Updated the ASB operator name based on what @shawn-hurley said he's changing it to.

shawn-hurley commented on modules/sb-start-asb.adoc:
```diff
@@ -0,0 +1,29 @@
+
+. Navigate in the web console to *Catalog* -> *Installed Operators* and select the `ansible-service-broker` project.
+. Select the *OpenShift Ansible Service Broker Operator*.
+. Under *Provided APIs*, click *Create New* for *OpenShift Ansible Broker*.
```
This won't change. Changing the CRD's that the operator watches is not something that we can do right now.

bergerhoffer replied:
@shawn-hurley Okay so this should still just say "Automation Broker"?

shawn-hurley commented:
We may have another step that needs to be taken to connect the OASB to the downstream registry. I will update once I know for sure :)

chrisnegus commented:
LGTM

bergerhoffer commented:
Merging this. If anything else comes up, we can address in a followup PR.

jboyd01 commented on modules/sb-install-service-catalog.adoc:
```diff
@@ -0,0 +1,56 @@
+
+.Procedure
+
+. Create the service catalog API server.
```
In recent builds, the ServiceCatalogAPIServer object is created during initial cluster creation, but the managed state is set to "Removed".  To enable Service Catalog, admins will need to  `oc edit ServiceCatalogAPIServer` and change the `managedState` from `Removed` to `Managed`.  This could be done either by using `oc edit` or using the file you have listed here and `oc apply -f svcat-api-server.yaml`  I imagine the latter is more in keeping with how we instruct admins to do things like this?

bergerhoffer replied:
I don't know if one way is preferred over the other, but I'll go with the file method for now and we can always update to use oc edit if we think that's the better method.

jboyd01 commented on modules/sb-install-service-catalog.adoc:
```diff
@@ -0,0 +1,56 @@
+.. Use the following command to create the service catalog API server.
++
+----
+$ oc create -f svcat-api-server.yaml
```
Similar to above with API Server, the ServiceCatalogControllerManager resource is created during install, but admins need to set the `managedStage` to `Managed`.  Change `oc create` to `oc apply`.

bergerhoffer replied:
@jboyd01 Actually a question - the files right now set "managementState" to Managed (for this and the API server. Is this correct, or should it be "managedStage" like your comments say?

jboyd01 commented on applications/service_brokers/installing-service-catalog.adoc:
```diff
@@ -0,0 +1,12 @@
+
+toc::[]
+
+IMPORTANT: The service catalog is deprecated in this release. Equivalent and better functionality is present in the Operator Framework and
```
Probably check with @mpbarrett or @robszumski on the exact wording here (ie Equivalent and better...)  I just want to be certain we are putting out the right message.  We use this term in the brokers and catalog section, so once you have the right text we'll want to update all occurrences.

bergerhoffer replied:
@jboyd01, @chengzhang1016:

@mpbarrett actually provided the wording to me for the deprecation notice for the release notes, see here: http://file.rdu.redhat.com/~ahardin/02252019/4-0-release-notes/release_notes/ocp-4-0-release-notes.html#ocp-4-0-notable-technical-changes

I pulled just part of this in to the deprecation notice in the actual content here. Let me know if this wording needs to be updated at all here or in the release notes. Thanks!

zihantang-rh commented on modules/sb-start-tsb.adoc:
```diff
@@ -0,0 +1,38 @@
+* Template service broker Pod status
+** From the *Workloads* -> *Pods* page for the *template-service-broker*
+project, verify that the Pod that starts with `apiserver-` has a status of
+*Running* and readiness of *Ready*.
```
Add checking of clusterservicebroker status. 
from UI:
Catalog-> Broker Management-> Service Brokers
status of 'template-service-broker ' should be ready.

zihantang-rh commented on modules/sb-install-asb-operator.adoc:
```diff
@@ -0,0 +1,42 @@
+.. Click *Create*.
+. Navigate to the *Catalog* -> *Operator Hub* page. Verify that the `ansible-service-broker` project is selected.
+. Select *OpenShift Ansible Service Broker Operator*.
++
```
This should be 'Automation Broker Operator'

chengzhang1016 replied:
At present, the name of asb in operator hub is 'Automation Broker Operator'. 
@shawn-hurley Hope double confirm with you.

chengzhang1016 replied:
As shawn-hurley said in other place, please ignore this comments at present. We will double confirm while downstream ready.

zihantang-rh commented on modules/common-attributes.adoc:
```diff
@@ -9,3 +9,4 @@
 :prewrap!:
 :op-system-first: Red Hat Enterprise Linux CoreOS (RHCOS)
 :op-system: RHCOS
+:asb-name: OpenShift Ansible broker
```
asb-name is 'Ansible Service Broker', this will be more accurate.

chengzhang1016 replied:
I remember we were using 'Openshift Ansible broker' as official name in last several OCP release docs. 
@shawn-hurley How do you think?

zihantang-rh commented on modules/sb-start-asb.adoc:
```diff
@@ -0,0 +1,29 @@
+[id='sb-start-asb-{context}']
+= Starting the {asb-name}
+
+Once you have installed the {asb-name} Operator, you can start the {asb-name} using the following procedure.
```
.... installed 'Automation Broker Operator'

chengzhang1016 replied:
As shawn-hurley said in other place, please ignore this comments at present. We will double confirm while downstream ready.

zihantang-rh commented on modules/sb-start-asb.adoc:
```diff
@@ -0,0 +1,29 @@
+
+* You have enabled service catalog.
+* You have installed the {asb-name} Operator.
+
```
'Automation Broker Operator'

chengzhang1016 replied:
As shawn-hurley said in other place, please ignore this comments at present. We will double confirm while downstream ready.

zihantang-rh commented on modules/sb-start-asb.adoc:
```diff
@@ -0,0 +1,29 @@
+.Procedure
+
+. Navigate in the web console to *Catalog* -> *Installed Operators* and select the `ansible-service-broker` project.
+. Select the *OpenShift Ansible Service Broker Operator*.
```
Automation Broker Operator.

shawn-hurley replied:
No, we are making the downstream one different than the upstream community CSV. We are changing these names.

chengzhang1016 replied:
OK. Got it. Thanks. @shawn-hurley

zihantang-rh commented on modules/sb-start-asb.adoc:
```diff
@@ -0,0 +1,29 @@
+
+* {asb-name} Pod status
+** From the *Workloads* -> *Pods* page for the *ansible-service-broker* project, verify that the Pod that starts with `asb-` has a status of *Running* and readiness of *Ready*.
+* Service catalog controller manager Pod logs
```
Add checking of clusterservicebroker status.
from UI:
Catalog-> Broker Management-> Service Brokers
status of 'ansible-service-broker ' should be ready.

Broker status will be ready a few minutes after the '=apiserver-' ready.

zihantang-rh commented on modules/sb-start-asb.adoc:
```diff
@@ -0,0 +1,29 @@
+If you do not yet see these Service classes, you can check the status of the following items:
+
+* {asb-name} Pod status
+** From the *Workloads* -> *Pods* page for the *ansible-service-broker* project, verify that the Pod that starts with `asb-` has a status of *Running* and readiness of *Ready*.
```
Add checking of clusterservicebroker status.
from UI:
Catalog-> Broker Management-> Service Brokers
status of 'ansible-service-broker ' should be ready.

Broker status will be ready a few minutes after the pod 'asb-' ready.

zihantang-rh commented:
@bergerhoffer sorry for review late about this PR. I add some comments, can you update this in the following up PRs . Thanks.

chengzhang1016 commented on applications/service_brokers/installing-ansible-service-broker.adoc:
```diff
@@ -0,0 +1,22 @@
+provides.
+
+IMPORTANT: The {asb-name} is deprecated in this release. Equivalent and better functionality is present in the Operator Framework and
+Operator Life Cycle Manager (OLM).
```
Samely concern with @jboyd01 , hope confirm that we are using exact word.
@shawn-hurley
